### PR TITLE
fix(core): add liveAnnouncer to message-toast service

### PR DESCRIPTION
**Pull request by OleksiyDrozdov**
closes [#5802](https://github.com/SAP/fundamental-ngx/issues/5802)

## Description

Now when a message toast appears on the screen, the information regarding the message toast along with the information available within the message toast is read even if the element does not have the focus.

##### During Implementation

1. Visual Testing:

-   [n/a] visual misalignments/updates
-   [n/a] check Light/Dark/HCB/HCW themes
-   [n/a] RTL/LTR - proper rendering and labeling
-   [n/a] responsiveness(resize)
-   [n/a] Content Density (Cozy/Compact/(Condensed))
-   [n/a] States - hover/disabled/focused/active/on click/selected/selected hover/press state
-   [n/a] Interaction/Animation - open/close, expand/collapse, add/remove, check/uncheck
-   [x] Mouse vs. Keyboard support
-   [n/a] Text Truncation

2. API and functional correctness

-   [x] check for console logs (warnings, errors)
-   [n/a] API boundary values
-   [n/a] different combinations of components - free style
-   [n/a] change the API values during testing

3. Documentation and Example validations

-   [n/a] missing API documentation or it is not understandable
-   [n/a] poor examples
-   [x] Stackblitz works for all examples

4. Accessibility testing
5. Browser Testing - Edge, Safari, Chrome, Firefox

### CODE REVIEW

File: apps/docs/src/app/core/component-docs/message-toast/examples/message-toast-example.component.ts [hunk 1]
```diff
@@ -8,28 +9,35 @@ import { MessageToastContentExampleComponent } from './message-toast-content-exa
...
+    constructor(public messageToastService: MessageToastService, public liveAnnouncer: LiveAnnouncer) {}
 
     openFromComponent(): void {
         this.messageToastService.open(MessageToastContentExampleComponent, {
             mousePersist: true,
             duration: 5000,
             maxWidth: '25rem'
         });
+        this.messageToastText = this.messageToastService.textMessage;
+        this.liveAnnouncer.announce(this.messageToastText);
```
Comment (platon-rov) [inline apps/docs/src/app/core/component-docs/message-toast/examples/message-toast-example.component.ts:23]: That's a great improvement!

But shouldn't we call live announcer's method from the library component, not from an example? To don't push customers to write such code in their applications.

Comment (OleksiyDrozdov) [reply apps/docs/src/app/core/component-docs/message-toast/examples/message-toast-example.component.ts]: done

**Review by platon-rov — APPROVED** [review]

**Review by InnaAtanasova — APPROVED** [review]

**Comment by Sanjay-1312** [comment]
Hello Colleagues,

Issue 1: Partially Fixed
image

Observed Behavior: The issue is partially fixed, On activating the buttons, ‘Open from component, open from template, open from string’ the message toast inside information is read but the message toast available information is not read.

 Expected Behavior: when button is activated the message toast  is available  should be read such as 'Alert' followed by message toast details information.

For reference:
https://sapui5untested.int.sap.eu2.hana.ondemand.com/#/entity/sap.m.MessageToast/sample/sap.m.sample.MessageToast

Retest Details:
https://deploy-preview-7103--fundamental-ngx.netlify.app/#/core/message-toast
JAWS 2021

**Review by mikerodonnell89 — APPROVED** [review]

**Comment by OleksiyDrozdov** [comment]
Hello @Sanjay-1312 

Could you please clarify Expected behaviour. Please see screenshot below how it is implemented now: when user presses button we can see pop up and screen reader is reading information from pop up. Should I just add word 'Alert' before text from pop up.

Screenshot 2021-11-09 220152
image

**Comment by Sanjay-1312** [comment]
Hello Colleague, 

The message toast is appearing on activating the buttons, The 'Alert' information should be read before the Message Toast information is being read.

Thanks and Regards
Sanjay

**Review by Betrozov — APPROVED** [review]

**Comment by puru-hk** [comment]
@OleksiyDrozdov Tested this issue message toast is reading twice. Please fix it it should read once. 

image

**Comment by OleksiyDrozdov** [comment]
> @OleksiyDrozdov Tested this issue message toast is reading twice. Please fix it it should read once.
> 
> image

done

**Comment by Sanjay-1312** [comment]
Hello Colleagues,

The reported issue is fixed.

Retest Details:

https://deploy-preview-7103--fundamental-ngx.netlify.app/#/core/message-toast
JAWS 2021

**Review by puru-hk — APPROVED** [review]

**Pull request merged**